external/netutils/netlib_getstats.c

```diff
@@ -0,0 +1,70 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+#include <netdb.h>
+#include <errno.h>
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: netlib_getstats
+ *
+ * Description:
+ *   Display current statistics informations of network stack.
+ *
+ * Returned Value:
+ *   0 on success, non-zero on failure
+ *
+ ****************************************************************************/
+
+int netlib_getstats(void)
+{
+	int ret = -1;
+	struct req_lwip_data req;
+
+	int sockfd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sockfd < 0) {
+		printf("socket() failed with errno: %d\n", errno);
+		return ret;
+	}
+
+	memset(&req, 0, sizeof(req));
+	req.type = GETNETSTATS;
+
+	ret = ioctl(sockfd, SIOCLWIP, (unsigned long)&req);
+	close(sockfd);
+	if (ret == ERROR) {
+		printf("ioctl() failed with errno: %d\n", errno);
+		close(sockfd);
+		return ret;
+	}
+
+	ret = req.req_res;
```

Comment:
simply return req.req_res;
But one question, It returns req.req_res, so description of this function is correct?